webapi/Controllers/MaintenanceController.cs

```diff
@@ -53,7 +53,7 @@ public class MaintenanceController : ControllerBase
 
         var migrationStatus = await migrationMonitor.GetCurrentStatusAsync(cancellationToken);
 
-        if (migrationStatus != ChatMigrationStatus.None)
+        if (migrationStatus == ChatMigrationStatus.Upgrading)
```

Comment:
Why change this? What about the RequiresUpgrade state?

Also, maybe we could simplify things and make that flag part of ServiceInfo instead of having to make a call for ServiceInfo and another for Maintenance status.

Reply:
because of this comment on the frontend:
<img width="459" alt="image" src="https://github.com/microsoft/chat-copilot/assets/52973358/9ee66929-a1bd-4a53-af94-cf7c9e2e0da2">

there is an existing assumption that when an object is returned from the `/maintenanceStatus` API, that the backend is in maintenance mode, but this is not the case when the status is `RequiresUpgrade` (which mine was in and is why I noticed this).

Reply:
I think "RequiresUpgrade" is also valid maintenance status.  Not sure how this change affects maintenance data-flow.

Reply:
Doesn't the `RequiresUpgrade` state trigger the upgrade automatically?

Reply:
I'm going to leave it as-is for now to not block the PR

Reply:
We should either revert this change or confirm that other ChatMigrationStatus states, like `RequiresUpgrade` don't have an effect, otherwise, this will cause a regression.

Reply:
sorry that's what I meant, that I will revert the change.. on the frontend, it doesn't currently make a big difference since the `isMaintenance` flag isn't set